com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/LaunchRequestHandler.java

```diff
@@ -113,64 +122,153 @@
             }
         }
 
+        logger.info("Trying to launch Java Program with options:\n" + String.format("main-class: %s\n", launchArguments.mainClass)
+                + String.format("args: %s\n", launchArguments.args)
+                + String.format("module-path: %s\n", StringUtils.join(launchArguments.modulePaths, File.pathSeparator))
+                + String.format("class-path: %s\n", StringUtils.join(launchArguments.classPaths, File.pathSeparator))
+                + String.format("vmArgs: %s", launchArguments.vmArgs));
+
         try {
-            StringBuilder launchLogs = new StringBuilder();
-            launchLogs.append("Trying to launch Java Program with options:\n");
-            launchLogs.append(String.format("main-class: %s\n", launchArguments.mainClass));
-            launchLogs.append(String.format("args: %s\n", launchArguments.args));
-            launchLogs.append(String.format("module-path: %s\n", StringUtils.join(launchArguments.modulePaths, File.pathSeparator)));
-            launchLogs.append(String.format("class-path: %s\n", StringUtils.join(launchArguments.classPaths, File.pathSeparator)));
-            launchLogs.append(String.format("vmArgs: %s", launchArguments.vmArgs));
-            logger.info(launchLogs.toString());
-
-            IDebugSession debugSession = DebugUtility.launch(
-                    vmProvider.getVirtualMachineManager(),
-                    launchArguments.mainClass,
-                    launchArguments.args,
-                    launchArguments.vmArgs,
-                    Arrays.asList(launchArguments.modulePaths),
-                    Arrays.asList(launchArguments.classPaths),
-                    launchArguments.cwd,
-                    envVars);
-            context.setDebugSession(debugSession);
-            context.setVmStopOnEntry(launchArguments.stopOnEntry);
-            context.setMainClass(parseMainClassWithoutModuleName(launchArguments.mainClass));
-
-            logger.info("Launching debuggee VM succeeded.");
-
-            ProcessConsole debuggeeConsole = new ProcessConsole(debugSession.process(), "Debuggee", context.getDebuggeeEncoding());
-            debuggeeConsole.onStdout((output) -> {
-                // When DA receives a new OutputEvent, it just shows that on Debug Console and doesn't affect the DA's dispatching workflow.
-                // That means the debugger can send OutputEvent to DA at any time.
-                context.sendEvent(Events.OutputEvent.createStdoutOutput(output));
-            });
-
-            debuggeeConsole.onStderr((err) -> {
-                context.sendEvent(Events.OutputEvent.createStderrOutput(err));
-            });
-            debuggeeConsole.start();
+            IVirtualMachineManagerProvider vmProvider = context.getProvider(IVirtualMachineManagerProvider.class);
+
+            if (context.supportsRunInTerminalRequest()
```

Comment:
Please consider extracting this branch to a new function. The current function is already big.